### PR TITLE
CircleCI monthly update bug fix + adding sysdig secure

### PR DESCRIPTION
**Pull request by RYadav95**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -57,6 +55,18 @@ jobs:
           name: Publish Docker Image to Docker Hub
           command: |
             echo "${DOCKERHUB_PASS}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+            TAG=${CIRCLE_TAG/v/''}
+            D_TAG="$(wget -q https://registry.hub.docker.com/v1/repositories/logdna/logdna-agent/tags -O -  | sed -e 's/[][]//g' -e 's/"//g' -e 's/ //g' | tr '}' '\n'  | awk -F: '{print $3}' | sort -g -r | head -1)"
+            lenght="$(echo $TAG | wc -m)"
+            if [ $lenght > 3 ]; then
```
Comment (smusali) [inline .circleci/config.yml]: why don't you just use `$(echo $TAG | wc -m)` instead of `$lenght`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -57,6 +55,18 @@ jobs:
...
+              IMAGE_TAG=$D_TAG
```
Comment (smusali) [inline .circleci/config.yml]: I'd suggest to use brackets for variables too: `${D_TAG}` instead of `$D_TAG` - this makes thing much cleaner and safer.

**Comment by baronomasia** [comment]
I'm confused as to the purpose of this PR. What are we trying to here exactly?

**Comment by RYadav95** [comment]
@baronomasia Monthly automatic build for LogDNA-agent was not working. For some reason git diff is not showing the change I made to fix the bug.
Added the image scanning feature as well. Using Sysdig secure for the image scan.

**Comment by baronomasia** [comment]
Ah ok, I see. Where is this automatic build being deposited after it is built?

**Comment by RYadav95** [comment]
So current workflow is
For a tag commit(ex. 1.5.6) circle will build a docker image. Push it to docker hub as logdna/logdna-agent:sandbox-{$DATESTAMP}. After that sysdig secure will scan the image.
After this, CircleCI will hold the pipeline for someone's approval. Once approved, CircleCI will publish the image to docker hub and ibm registry with appropriate tags.
This pipeline will also trigger at the beginning of every month.

**Comment by RYadav95** [comment]
@smusali  I made all the necessary changes. Please review and merge it.

File: .circleci/config.yml [hunk 3]
```diff
@@ -57,6 +55,17 @@ jobs:
           name: Publish Docker Image to Docker Hub
           command: |
             echo "${DOCKERHUB_PASS}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+            TAG=${CIRCLE_TAG}
+            D_TAG="$(wget -q https://registry.hub.docker.com/v1/repositories/logdna/logdna-agent/tags -O -  | sed -e 's/[][]//g' -e 's/"//g' -e 's/ //g' | tr '}' '\n'  | awk -F: '{print $3}' | sort -g -r | head -1)"
```
Comment (gregswift) [inline .circleci/config.yml]: You might want to try this method. Reduces the pipes and ignores tags that shouldnt be relevant. It also restricts this to just the v1 in this repository with the MAJOR_VERSION grep.  You'll want to add this up at the top. I'll commment there too.

```D_TAG="$(wget -q https://registry.hub.docker.com/v1/repositories/logdna/logdna-agent/tags -O - | jq -r '.[] | .name' | awk '!/-/ && /\./ { print $1}' | grep ${MAJOR_VERSION} | tail -1)"```

Comment (gregswift) [reply .circleci/config.yml]: On line 13 you would want to add the `MAJOR_VERSION=1`

I also think replacing latest everywhere with a `BASE_TAG=sandbox` would probably be a good idea for simplicity and chnges in the future.

File: .circleci/config.yml [hunk 4]
```diff
@@ -57,6 +56,18 @@ jobs:
           name: Publish Docker Image to Docker Hub
           command: |
             echo "${DOCKERHUB_PASS}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+            TAG=${CIRCLE_TAG}
+            D_TAG="$(wget -q https://registry.hub.docker.com/v1/repositories/logdna/logdna-agent/tags -O - | jq -r '.[] | .name' | awk '!/-/ && /\./ { print $1}' | grep ${MAJOR_VERSION} | tail -1)"d
```
Comment (gregswift) [inline .circleci/config.yml]: There is a `d` at the end of this line. Copy pasta or go issue I think

Comment (baronomasia) [reply .circleci/config.yml]: maybe it's intentional? the var name is D_TAG and this certainly would put a d character in the tag string...

**Pull request merged**